src/ol/interaction/Select.js

```diff
@@ -526,12 +526,23 @@ class Select extends Interaction {
          */
         function (feature, layer) {
           if (this.filter_(feature, layer)) {
-            if ((add || toggle) && !includes(features.getArray(), feature)) {
+            if (
+              (add || toggle) &&
+              !features
+                .getArray()
+                .includes(
+                  /** @type {import("../Feature.js").default} */ (feature)
```

Comment:
Isn't this showing a problem with the type returned by `getFeatures`, and its return type should be `Array<FeatureLike>`?

Reply:
Currently digging in that can of worms.  Not clear to me where this interaction actually works with render features.  There are parts that assume a full fledged feature.

Reply:
I think that ae87cad7ef65dbc8b7cf490809e17ee891961a4b added a few too many `FeatureLike` types.  The select interaction only really works with `Feature` instances.  I pushed a new commit that makes this clear.